substrate/client/network/src/litep2p/shim/request_response/mod.rs

```diff
@@ -371,10 +373,33 @@ impl RequestResponseProtocol {
 
 		let status = match error {
 			RequestResponseError::NotConnected =>
-				Some((RequestFailure::NotConnected, "not-connected")),
-			RequestResponseError::Rejected => Some((RequestFailure::Refused, "rejected")),
+				Some((RequestFailure::NotConnected, "not-connected".to_string())),
+			RequestResponseError::Rejected(reason) => {
+				let reason = match reason {
+					RejectReason::ConnectionClosed => "connection-closed".to_string(),
+					RejectReason::SubstreamClosed => "substream-closed".to_string(),
+					RejectReason::SubstreamOpenError(substream_error) => {
+						format!("substream-open-error: {:?}", substream_error)
```

Comment:
Do we really want this as a metric name?

Reply:
Yep that's a good point, have remove the String allocation entirely and provided shorter `str` names, thanks 🙏